### PR TITLE
Repl BQ macros

### PR DESCRIPTION
**Pull request by ravwojdyla**

### CODE REVIEW

**Comment by ravwojdyla** [comment]
This one is based on #20 - extension with BQ marcos work in REPL.

File: scio-repl/README.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+# Quick start
+
+The easiest way to start with Scio REPL is to assembly jar and run it:
+
+```bash
+$ sbt 'project scio-repl' assembly
+$ java -jar scio-repl/target/scala-2.11/scio-repl*-fat.jar
+Starting up ...
+Scio context is available at 'sc'
```
Comment (nevillelyh) [inline scio-repl/README.md]: `ScioContext is available as 'sc'` everywhere?

File: scio-repl/README.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+```
+Scio context is available as `sc`, this is the starting point, use `tab` completion, history and
+all the other goods of REPL to play around in local mode.
+
+# Tutorial
+
+Start simple pipeline from
+
+```
+scio> val noTwoS = sc.parallelize(List(1,2,3)).filter( _ != 2 ).map( "I like " + _ ).saveAsTextFile("/tmp/hate-2s")
```
Comment (nevillelyh) [inline scio-repl/README.md]: WordCount with `README.md` instead? Makes more sense?

File: scio-repl/README.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+[main] INFO com.google.cloud.dataflow.sdk.runners.DirectPipelineRunner - Pipeline execution complete.
+result: com.spotify.scio.ScioResult = com.spotify.scio.ScioResult@220a5163
+scio> result.state
+res20: com.google.cloud.dataflow.sdk.PipelineResult.State = DONE
+scio> noTwoS.waitForResult().value.foreach(println)
+I like 3
+I like 1
+```
+
+To create Scio context connected to Google's Dataflow service pass Scio arguments on REPL startup -
```
Comment (nevillelyh) [inline scio-repl/README.md]: To create a ScioContext for Google Cloud Dataflow service pass Dataflow pipeline options ...?

File: scio-repl/README.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+```bash
+$ java -jar scio-repl/target/scala-2.11/scio-repl*-fat.jar \
+> --project=<project-id> \
+> --stagingLocation=<stagin-dir> \
+> --runner=BlockingDataflowPipelineRunner
+Starting up ...
+Scio context is available at 'sc'
+Welcome to Scio REPL!
+scio> sc.parallelize(List(1,2,3)).map( _.toString ).saveAsTextFile("gs://<output-dir>")
```
Comment (nevillelyh) [inline scio-repl/README.md]: Use shakespeare/kinglear.txt as the official WC example?

File: scio-repl/README.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+While in the REPL, use `:paste` magic command to past/write multi line code
+
+```
+scio> :paste
+// Entering paste mode (ctrl-D to finish)
+def myWordFilter(word: String) = word.startsWith("rav")
+val importantNames = sc.parallelize
+parallelize   parallelizeTimestamped
+val importantNames = sc.parallelize(List("ravioli", "rav", "jbx")).filter(myWordFilter)
```
Comment (nevillelyh) [inline scio-repl/README.md]: Clean up the tab-completion printouts and do a BQ macros example?

File: scio-repl/README.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+// Exiting paste mode, now interpreting.
+myWordFilter: (word: String)Boolean
+importantNames: com.spotify.scio.values.SCollection[String] = com.spotify.scio.values.SCollectionImpl@75882ac2
+scio> importantNames.saveAsTextFile("/tmp/ravs")
+scio> sc.close
+```
+
+## `DataflowPipelineRunner` = more interactive Dataflow service
+
+When using REPL and Dataflow service consider using `DataflowPipelineRunner` to get more
```
Comment (nevillelyh) [inline scio-repl/README.md]: `using the non-blocking.`

File: scio-repl/README.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+res3: Boolean = false
+scio> :newScio nextSc
+```
+
+Keep in mind tho that currently `BlockingDataflowPipelineRunner` is still pretty noise on stdout,
+thus, currently, it's recommended to use `DataflowPipelineRunner` with `asyncClose`.
+
+## Use BigQuery macros
+
+BigQuery macros allows for typesafe and civilized integration with BigQuery inside Scio. Use it
```
Comment (nevillelyh) [inline scio-repl/README.md]: `@BigQueryType annotations enable type safe and...`?

File: scio-repl/README.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+
+```bash
+java -jar -Dbigquery.project=<project-id> scio-repl/target/scala-2.11/scio-repl*-fat.jar
+Starting up ...
+Scio context is available at 'sc'
+Welcome to Scio REPL!
+scio> @BigQueryType.fromQuery("SELECT tornado, month FROM [publicdata:samples.gsod]") class Row
+defined class Row
+defined object Row
+```
```
Comment (nevillelyh) [inline scio-repl/README.md]: Will be nice to add BQ direct query without Scio/DF. E.g.
```
import com.spotify.scio.bigquery._
val bq = BigQueryClient()
bq.getTypedRows[Row]
```

**Pull request merged**